### PR TITLE
Added the Amplitude Rescaling Transform

### PR DESCRIPTION
**Pull request by payo101**
**Closes**: https://github.com/lightly-ai/lightly/issues/1658

**Modified file(s)**
- `lightly/transforms/__init__.py`: Added new transform imports

**Created file(s)**
- `lightly/transforms/amplitude_rescale_transform.py`
- `tests/transforms/test_amplitude_rescale_transform.py`

### CODE REVIEW

**Comment by payo101** [comment]
Sure, @SauravMaheshkar , give me some time on that

File: tests/transforms/test_amplitude_rescale_transform.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    rfftTransform = RFFT2DTransform()
+    rfft = rfftTransform(image)
+
+    ampRescaleTf_1 = AmplitudeRescaleTranform()
+    rescaled_rfft_1 = ampRescaleTf_1(rfft)
+
+    ampRescaleTf_2 = AmplitudeRescaleTranform(range=(1.0, 2.0))
+    rescaled_rfft_2 = ampRescaleTf_2(rfft)
+
+    assert rescaled_rfft_1.shape == rfft.shape and rescaled_rfft_2.shape == rfft.shape
```
Comment (guarin) [inline tests/transforms/test_amplitude_rescale_transform.py:21]:
```suggestion
    assert rescaled_rfft_1.shape == rfft.shape
    assert rescaled_rfft_2.shape == rfft.shape
```

File: lightly/transforms/amplitude_rescale_transform.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import numpy as np
+import torch
+from torch import Tensor
+
+
+class AmplitudeRescaleTranform:
+    """
+    This transform will rescale the amplitude of the Fourier Spectrum (`input`) of the image and return it.
+    The scaling value *p* will range within `[m, n)`
```
Comment (guarin) [inline lightly/transforms/amplitude_rescale_transform.py:11]: Please follow the same docstring formatting as for other transforms: https://github.com/lightly-ai/lightly/blob/b8613dc6765e26e7bc19fb99797a26656e6e389d/lightly/transforms/rotation.py#L13-L31

File: lightly/transforms/amplitude_rescale_transform.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        **input**: _torch.Tensor_
+        The 2D Discrete Fourier Tranform of an Image.
+    # Returns:
+        **output**:_torch.Tensor_
+        The Fourier spectrum of the 2D Image with rescaled Amplitude.
+    """
+
+    def __init__(self, range: Tuple[float, float] = (0.8, 1.75)) -> None:
+        self.m = range[0]
+        self.n = range[1]
```
Comment (guarin) [inline lightly/transforms/amplitude_rescale_transform.py:35]: Uniform can be imported from `from torch.distributions import Uniform`

```suggestion
        self.dist = Uniform(range[0], range[1])
```

File: lightly/transforms/amplitude_rescale_transform.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    def __call__(self, input: Tensor) -> Tensor:
```
Comment (guarin) [inline lightly/transforms/amplitude_rescale_transform.py:37]: Let's call it `freq_image` as in the other transforms.
```suggestion
    def __call__(self, freq_image: Tensor) -> Tensor:
```

File: lightly/transforms/amplitude_rescale_transform.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        p = np.random.uniform(self.m, self.n)
+
+        output = input * p
+
```
Comment (guarin) [inline lightly/transforms/amplitude_rescale_transform.py:41]: The noise should only be added to the amplitude. For this the image has to be decomposed into amplitude and phase: 

```suggestion
        amplitude = torch.sqrt(freq_image.real ** 2 + freq_image.imag ** 2)
        phase = torch.atan2(freq_image.imag, freq_image.real)
        # p with shape (H, W)
        p = self.dist.sample(freq_image.shape[1:]).to(freq_image.device)
        # Unsqueeze to add channel dimension.
        amplitude *= p.unsqueeze(0)
        real = amplitude * torch.cos(phase)
        imag = amplitude * torch.sin(phase)
        output = torch.complex(real, imag)
```

**Review by guarin — COMMENTED** [review]
Thanks for the contribution! Left some comments.

**Comment by payo101** [comment]
Hi @guarin , sorry being so late on this. I have made the requested changes. Please have a look

**Pull request merged**